.buildkite/dagster-buildkite/dagster_buildkite/package_build_spec.py

```diff
@@ -0,0 +1,247 @@
+import os
+from typing import Callable, List, Mapping, NamedTuple, Optional, Union
+
+from .python_version import AvailablePythonVersion
+from .step_builder import BuildkiteQueue
+from .steps.tox import build_tox_step
+from .utils import BuildkiteLeafStep, GroupStep
+
+_CORE_PACKAGES = [
+    "python_modules/dagster",
+    "python_modules/dagit",
+    "python_modules/dagster-graphql",
+    "js_modules/dagit",
+]
+
+
+def _infer_package_type(directory: str) -> str:
+    if directory in _CORE_PACKAGES:
+        return "core"
+    elif directory.startswith("examples/"):
+        return "example"
+    elif directory.startswith("python_modules/libraries/"):
+        return "extension"
+    else:
+        return "infrastructure"
+
+
+_PACKAGE_TYPE_TO_EMOJI_MAP: Mapping[str, str] = {
+    "core": ":dagster:",
+    "example": ":large_blue_diamond:",
+    "extension": ":electric_plug:",
+    "infrastructure": ":gear:",
+}
+
+PytestExtraCommandsFunction = Callable[[AvailablePythonVersion, Optional[str]], List[str]]
+PytestDependenciesFunction = Callable[[AvailablePythonVersion, Optional[str]], List[str]]
+
+
+class PackageBuildSpec(
```

Comment:
nit: 
`PythonPackageTestSpec` 
`PackageTestSpec` 

since these aren't really "build" steps

Reply:
Good point-- maybe just `PackageSpec`? Since we're both testing and linting, and maybe something else in the future too.